tests/test_telemetry.py

```diff
@@ -598,146 +580,83 @@ class Test_TelemetryConfigurationChaining:
                     {"name": "DD_LOGS_INJECTION", "origin": "code", "value": True},
                 ],
             },
-        },  # configurations should be ordered according to _ORIGIN_PRECEDENCE_ORDER
+        },
+        "python": {
+            "configuration": {
+                "DD_LOGS_INJECTION": [
+                    {"name": "DD_LOGS_INJECTION", "origin": "default", "value": True},
+                    {"name": "DD_LOGS_INJECTION", "origin": "env_var", "value": False},
+                    {"name": "DD_LOGS_INJECTION", "origin": "code", "value": True},
+                ],
+            },
+        },
     }
 
-    @classmethod
-    def get_origin_precedence_order(cls) -> list[str]:
-        """Get the official configuration origin precedence order (lowest to highest)
-
-        Returns:
-            List of origin names in precedence order from lowest to highest
-
-        """
-        return cls._ORIGIN_PRECEDENCE_ORDER.copy()
-
     @scenarios.telemetry_app_started_config_chaining
-    def test_app_started_config_chaining(self):
-        """Assert that all configuration sources read at start time are sent with the app-started event in the correct order"""
-        self._validate_config_chaining(require_seq_id=True)
-
-    @scenarios.telemetry_app_started_config_chaining
-    def test_app_started_config_chaining_no_seq_id(self):
-        """If for some reason the seq_id is not sent, assert that all configuration sources read at start time for a given configuration
-        are sent with the app-started event in the correct order
-        """
-        self._validate_config_chaining(require_seq_id=False)
-
-    def _validate_config_chaining(self, *, require_seq_id: bool = True) -> None:
-        """Common validation logic for configuration chaining tests
-
-        Args:
-            require_seq_id: Whether to require and validate seq_id presence and ordering
-
-        """
-        nodejs_expected_config = self._CONFIG_CHAINING_TEST_CONFIG["nodejs"]["configuration"]
+    def test_app_started_config_chaining_seq_id(self):
+        """Assert that the seq_id is sent for each configuration entry in the app-started event"""
 
         def validator(data):
-            if get_request_type(data) != "app-started":
+            if get_request_type(data) not in ["app-started", "app-client-configuration-change"]:
                 return
 
             content = data["request"]["content"]
             configurations = content["payload"]["configuration"]
+            assert configurations, f"No configurations found in app-started event: {configurations}"
```

Comment:
The test name and the assertion message indicate that this test is looking explicitly for app-started event, but the change to line 600 says otherwise.